### ISSUE TITLE
runPluginVerifier: cannot use IU-2021.1 EAP

### ISSUE THREAD

**Issue by L2jLiga** [JetBrains/gradle-intellij-plugin#606]
Hello,

When `IU-2021.1` is used in plugin verifier IDEs list it fails with error, see [Github Actions step](https://github.com/L2jLiga/fastify-decorators-plugin/runs/1967895716?check_suite_focus=true#step:8:630), however I was able to use same approach for other versions (e.g. `IU-2020.3.2`).

When I switched to `IU-211.6085.26` it works fine, but now fails on `IU-2020.2.4` [Github Actions step](https://github.com/L2jLiga/fastify-decorators-plugin/runs/1968131278?check_suite_focus=true#step:8:429)

Also I have same issue with WS 2021.1 EAP even when specify build explicitely (WS-211.6085.22), see [Github Actions step](https://github.com/L2jLiga/fastify-decorators-plugin/runs/1967953860?check_suite_focus=true#step:8:788)

plugin version is `0.7.2`

**Comment by mvilliger** [comment]
We have the same issue with gradle-intellij-plugin version 0.8.0-SNAPSHOT:

```
Task ':runPluginVerifier' is not up-to-date because:
  No history is available.
[gradle-intellij-plugin :myPlugin:runPluginVerifier] Downloaing IDE: IU-2020.2.4
Download https://cache-redirector.jetbrains.com/download.jetbrains.com/idea/ideaIU-2020.2.4.tar.gz
[gradle-intellij-plugin :myPlugin:runPluginVerifier] Cannot resolve direct download URL for: https://data.services.jetbrains.com/products/download?code=IU&platform=linux&type=rc&version=2020.2.4
java.io.FileNotFoundException: https://data.services.jetbrains.com/products/download?code=IU&platform=linux&type=rc&version=2020.2.4
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1920)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1520)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:250)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl$getInputStream$0.call(Unknown Source)
	at org.jetbrains.intellij.tasks.RunPluginVerifierTask.resolveIdeUrl(RunPluginVerifierTask.groovy:705)
	at org.jetbrains.intellij.tasks.RunPluginVerifierTask.downloadIde(RunPluginVerifierTask.groovy:655)
	...
[gradle-intellij-plugin :myPlugin:runPluginVerifier] Cannot resolve direct download URL for: https://data.services.jetbrains.com/products/download?code=IU&platform=linux&type=eap&version=2020.2.4
java.io.FileNotFoundException: https://data.services.jetbrains.com/products/download?code=IU&platform=linux&type=eap&version=2020.2.4
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1920)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1520)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:250)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl$getInputStream$0.call(Unknown Source)
	at org.jetbrains.intellij.tasks.RunPluginVerifierTask.resolveIdeUrl(RunPluginVerifierTask.groovy:705)
	at org.jetbrains.intellij.tasks.RunPluginVerifierTask.downloadIde(RunPluginVerifierTask.groovy:655)
	...
[gradle-intellij-plugin :myPlugin:runPluginVerifier] Cannot resolve direct download URL for: https://data.services.jetbrains.com/products/download?code=IU&platform=linux&type=beta&version=2020.2.4
java.io.FileNotFoundException: https://data.services.jetbrains.com/products/download?code=IU&platform=linux&type=beta&version=2020.2.4
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1920)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1520)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:250)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl$getInputStream$0.call(Unknown Source)
	at org.jetbrains.intellij.tasks.RunPluginVerifierTask.resolveIdeUrl(RunPluginVerifierTask.groovy:705)
	at org.jetbrains.intellij.tasks.RunPluginVerifierTask.downloadIde(RunPluginVerifierTask.groovy:655)
    ...

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':runPluginVerifier'.
> IDE 'IU-2020.2.4' cannot be downloaded. Please verify the specified IDE version against the products available for testing: https://jb.gg/intellij-platform-builds-list

* Try:
Run with --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':runPluginVerifier'.
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at org.jetbrains.intellij.tasks.RunPluginVerifierTask.resolveIdePath(RunPluginVerifierTask.groovy:627)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.jetbrains.intellij.tasks.RunPluginVerifierTask$_runPluginVerifier_closure1.doCall(RunPluginVerifierTask.groovy:524)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.jetbrains.intellij.tasks.RunPluginVerifierTask.runPluginVerifier(RunPluginVerifierTask.groovy:524)
	...
Caused by: org.gradle.api.GradleException: IDE 'IU-2020.2.4' cannot be downloaded. Please verify the specified IDE version against the products available for testing: https://jb.gg/intellij-platform-builds-list
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at org.jetbrains.intellij.tasks.RunPluginVerifierTask.resolveIdePath(RunPluginVerifierTask.groovy:627)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.jetbrains.intellij.tasks.RunPluginVerifierTask$_runPluginVerifier_closure1.doCall(RunPluginVerifierTask.groovy:524)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.jetbrains.intellij.tasks.RunPluginVerifierTask.runPluginVerifier(RunPluginVerifierTask.groovy:524)
	...

* Get more help at https://help.gradle.org

BUILD FAILED in 6m 41s
[gradle-intellij-plugin :myPlugin:runPluginVerifier] Downloaing IDE: IU-2020.2.4
Download https://data.services.jetbrains.com/products/download?code=IU&platform=linux&type=rc&version=2020.2.4
[gradle-intellij-plugin :myPlugin:runPluginVerifier] Downloaing IDE: IU-2020.2.4
Download https://data.services.jetbrains.com/products/download?code=IU&platform=linux&type=eap&version=2020.2.4
[gradle-intellij-plugin :myPlugin:runPluginVerifier] Downloaing IDE: IU-2020.2.4
Download https://data.services.jetbrains.com/products/download?code=IU&platform=linux&type=beta&version=2020.2.4

> Task :runPluginVerifier FAILED
```

**Comment by L2jLiga** [comment]
From my end would like to add that issue appear when 3 or more IDEs specified in verifier list, does it helps @hsz ?

**Comment by hsz** [comment]
Sorry folks for the late reply – last week, I've finally found the root cause of the `IDE must reside in a directory` issue.
The cause of that issue was Plugin Verifier, which had a clearing cache feature available if the downloaded IDEs cache directory exceeded 10GB.
With [`v1.256`](https://github.com/JetBrains/intellij-plugin-verifier/releases/tag/v1.256), the issue has been fixed. This version was released both to the Bintray and Space Packages, so it should be available for everyone already (if no PV version was specified, it refers to the latest available).
Could you please verify if everything works fine for you?

**Comment by L2jLiga** [comment]
Yeah, @hsz I can confirm that issue is solved, I returned versions back to verifier list and CI is green. 

cc @mvilliger what about you? 😄

**Comment by mvilliger** [comment]
Thank you @hsz for looking into this.
Unfortunately I still get the same exceptions as in my previous post. I am using gradle-intellij-plugin version 0.8.0-SNAPSHOT to run the plugin verifier. Is another snapshot required? Or how can I adjust the plugin verifier used?

**Comment by L2jLiga** [comment]
Probably you need to cleanup your cache to ensure latest plugin verifier version is used

**Comment by mvilliger** [comment]
@L2jLiga The build is executed in a docker container which starts with an empty user home each time. So there should not be any cache available?

**Comment by L2jLiga** [comment]
Not sure, I agree, there should not be any cache, however I use plugin version 0.7.2 and running in Github Actions, my latest builds are green (https://github.com/L2jLiga/fastify-decorators-plugin/actions)

**Comment by L2jLiga** [comment]
Anyway your case is quite interesting, according to [products list](https://data.services.jetbrains.com/products?fields=code,name,releases.downloads,releases.version,releases.build,releases.type&code=IIU) I cannot find 2020.2.4 RC, Beta or EAP, only release, however first time it seems like plugin verifier tries to download correct version from https://cache-redirector.jetbrains.com/download.jetbrains.com/idea/ideaIU-2020.2.4.tar.gz and after that trying to download same version but with different codes... 

long story short: i'm confused why this may happens and it seems like to be slightly different issue

**Comment by mvilliger** [comment]
I totally agree. The first download to 2020.2.4 should succeed. No clue why the verifier tries the RC, Beta and eap version afterwards.
Maybe it is a different issue. At first i thought it may be related.

**Comment by L2jLiga** [comment]
Let's post this as separated issue then
